firmware/src/dimmer_model.c

```diff
@@ -203,8 +230,99 @@ bool ResetToBlockAddress(uint16_t start_address) {
   return true;
 }
 
+uint8_t *AddStatusMessageToResponse(uint8_t *ptr,
+                                    const StatusMessage *message) {
+  ptr = PushUInt16(ptr, message->sub_device);
+  *ptr++ = message->status_type;
+  ptr = PushUInt16(ptr, message->message_id);
+  ptr = PushUInt16(ptr, message->data_value1);
+  ptr = PushUInt16(ptr, message->data_value2);
+  return ptr;
+}
+
+void QueueStatusMessage(DimmerSubDevice *device,
+                        RDMStatusType status_type,
+                        RDMStatusMessageId status_id,
+                        uint16_t data_value1,
+                        uint16_t data_value2) {
+  if (device->sd_report_threshold == STATUS_NONE &&
+      (status_type & 0xf) < device->sd_report_threshold) {
+    return;
+  }
+
+  device->status_message.is_active = true;
+  device->status_message.sub_device = device->index;
+  device->status_message.status_type = status_type;
+  device->status_message.message_id = status_id;
+  device->status_message.data_value1 = data_value1;
+  device->status_message.data_value2 = data_value2;
+}
+
 // Root PID Handlers
 // ----------------------------------------------------------------------------
+int DimmerModel_GetStatusMessages(const RDMHeader *header,
+                                  UNUSED const uint8_t *param_data) {
+  uint8_t threshold = param_data[0];
+
+  if (threshold > STATUS_ERROR) {
+    return RDMResponder_BuildNack(header, NR_DATA_OUT_OF_RANGE);
+  }
+
+  uint8_t *ptr = g_rdm_buffer + sizeof(RDMHeader);
+
+  if (threshold == STATUS_GET_LAST_MESSAGE) {
+    // Return the last ones
```

Comment:
You can pack many status messages into a response. Maybe you're thinking of queued messages?